fspiop-api/documents/Tracing v1.0.md

```diff
@@ -0,0 +1,221 @@
+
+This section introduces Distributed Tracing from a Mojaloop context, including the header format, and the mechanisms used to generate, continue and forward traces.
+
+### 3.1 Tracing Data Model
+
+The [Table 1](#table-1) describes the trace headers fields that are optionally included in each Interoperability API request. Tables [2](#table-2), [3](#table-3), and [4](#table-4) describes the associated values and properties that is contained by each of the respective trace header fields shown in [Table 1](#table-1).
+
+**Note:** ...
+
+###### Table 1
+
+| Name | Cardinality | Type | Description | Reference |
+| --- | --- | --- | --- | --- |
+| traceparent | 1 | String(55) | Incoming request containing the trace-information, including the _version_ and populated _version-format_ values.<br/>Format: _{version}_-_{version-format}_.<br/>Example: `00-4bf92f3577b34da6a3ce929d0e0e4736-00f067aa0ba902b7-01` | Sections _[3.2.2 traceparent Header Field Values](https://www.w3.org/TR/trace-context-1/#traceparent-header-field-values)_ <sup>[1](https://www.w3.org/TR/trace-context-1/#traceparent-header-field-values)</sup> |
+| tracestate | 0...1 | String(512), <br/>list-member-keyvalue(0...32) | Provides optional vendor-specific trace information, and support for multiple distributed traces. It must be propagated when received, if and only if the value contains non-empty list (max of 32)  of _list-member-keyvalue_'s (Refer to [Table 4](#table-4)) that are comma (`,`) separated.<br/>Format: _{list-member-keyvalue}_,_{list-member-keyvalue}_,_{list-member-keyvalue}_. If the value of the traceparent field wasn't changed before propagation, tracestate MUST NOT be modified as well. <br/>Example: `vendorname1=opaqueValue1,vendorname2=opaqueValue2` | Section _[3.3 Tracestate Header](https://www.w3.org/TR/trace-context-1/#tracestate-header)_ <sup>[1](https://www.w3.org/TR/trace-context-1/#tracestate-header)</sup> |
+
+**Table 1 – Data model of HTTP header fields for Tracing**
+
+###### Table 2
+
+| Name | Cardinality | Type | Description | Reference |
+| --- | --- | --- | --- | --- |
+| version | 1 | 2HEXDIGLC | Version indicator for trace-format. Refer to [Table 3](#table-3) for supported versions. | Section _[3.2.2.1 version](https://www.w3.org/TR/trace-context-1/#version)_ <sup>[1](https://www.w3.org/TR/trace-context-1/#version)</sup> |
+| version-format | 1 | String(52) | A _dash_ (`-`) delimitated string included tracing information such as: trace-id, parent-id, trace-flags, etc. | Refer to [Table 3](#table-3) |
+| trace-id | 1 | 32HEXDIGLC | ID representing the entire trace, i.e. the end-to-end trace identifier. | Section _[3.2.2.3 trace-id](https://www.w3.org/TR/trace-context-1/#trace-id)_ <sup>[1](https://www.w3.org/TR/trace-context-1/#trace-id)</sup> |
+| parent-id | 1 | 16HEXDIGLC | ID representing the span-id.  | Section _[3.2.2.4 parent-id](https://www.w3.org/TR/trace-context-1/#parent-id)_ <sup>[1](https://www.w3.org/TR/trace-context-1/#parent-id)</sup> |
```

Comment:
`span-id` isn't defined anywhere. Maybe we need an image or something to help convey these different terms.

Reply:
Those terms are already defined in the w3 specification. I am not sure it's a good idea to replicate them here?

I.e. click the reference link and it will take you to the following:

----
3.2.2.4 parent-id
This is the ID of this request as known by the caller (in some tracing systems, this is known as `the span-id, where a span is the execution of a client request`). It is represented as an 8-byte array, for example, 00f067aa0ba902b7. All bytes as zero (0000000000000000) is considered an invalid value.

Reply:
I am happy to include a diagram, perhaps something similar to the following: https://docs.mojaloop.io/documentation/mojaloop-technical-overview/event-framework/#23-event-flow.

Reply:
I have updated table 2 to include the definition of a span for _parent-id_'s description.